legion/engine/core/particles/particlesystem.cpp

```diff
@@ -75,33 +77,33 @@ namespace legion::core
         auto& ageBuffer = emitter.get_buffer<life_time>("lifetimeBuffer");
         size_type destroyed = 0;
         size_type activeCount = 0;
-        for (activeCount = 0; activeCount < emitter.currentParticleCount; activeCount++)
+        for (activeCount = 0; activeCount < emitter.m_particleCount; activeCount++)
         {
             if (!emitter.is_alive(activeCount))
                 break;
 
             auto& lifeTime = ageBuffer[activeCount];
             lifeTime.age += deltaTime;
         }
-        if (!emitter.infinite)
+        if (emitter.has_uniform<float>("minLifeTime") && emitter.has_uniform<float>("maxLifeTime"))
         {
             for (size_type i = 0; i < activeCount; i++)
             {
                 auto& lifeTime = ageBuffer[i];
                 if (lifeTime.age > lifeTime.max)
                 {
                     emitter.set_alive(i, false);
-                    emitter.swap(i, emitter.currentParticleCount - 1);
-                    emitter.currentParticleCount--;
+                    emitter.swap(i, emitter.m_particleCount - 1);
+                    emitter.m_particleCount--;
                     destroyed++;
```

Comment:
maybe policies should have access to set alive, and set alive and swapping should be bundled together